homeassistant/components/fritz/common.py

```diff
@@ -134,9 +135,8 @@ def _update_info(self):
         """Retrieve latest information from the FRITZ!Box."""
         return self.fritzhosts.get_hosts_info()
 
-    def scan_devices(self, now: Optional[datetime] = None) -> None:
+    def scan_devices(self, now: datetime | None) -> None:
```

Comment:
This looks like a bug. This method is called without arguments in few places.

Reply:
I think this is by design as the time will come when its callback from async_track_interval, its just unused

Reply:
I mean this PR removed default value of `now` argument which seems wrong. And definetely has nothing with fixing pylint warnings.

Reply:
> I mean this PR removed default value of `now` argument which seems wrong. And definetely has nothing with fixing pylint warnings.

😱  you are spot on. This PR did introduce a bug.

Reply:
Thanks for spotting this one. Didn't catch that one. I opened #49668 to fix it